File: app/src/main/java/com/maple/recordwav/base/BaseFragment.java
package com.maple.recordwav.base;

import android.content.Context;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @author maple
 * @time 16/4/13 下午4:40
 */
public abstract class BaseFragment extends Fragment {
    public View view;
    public Context mContext;
    public FragmentManager fm;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.mContext = getActivity();
        this.fm = getFragmentManager();
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = initView(inflater);
        view.setClickable(true);
        return view;
    }


    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        initData(savedInstanceState);
        initListener();
        super.onActivityCreated(savedInstanceState);
    }


    /**
     * onCreateView方法中构建UI(将xml转换成view对象)
     */
    public abstract View initView(LayoutInflater inflater);

    /**
     * onActivityCreated方法中请求网络。返回数据填充UI
     */
    public abstract void initData(Bundle savedInstanceState);

    /**
     * onActivityCreated方法处理监听
     */
    public abstract void initListener();


}

File: app/src/main/java/com/maple/recordwav/base/BaseFragment.kt
package com.maple.recordwav.base

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

/**
 * base fragment
 *
 * @author maple
 * @time 2018/12/4
 */
abstract class BaseFragment : Fragment() {
    lateinit var mContext: Context

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mContext = context
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        var view = inflater.inflate(getLayoutRes(), container, false)
        view.isClickable = true // prevent click penetration
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView(view, savedInstanceState)
    }

    open fun onKeyBackPressed(): Boolean {
        return false
    }

    abstract fun getLayoutRes(): Int

    abstract fun initView(view: View, savedInstanceState: Bundle?)
}